include/aspect/particle/particle_handler.h

```diff
@@ -144,9 +199,60 @@ namespace aspect
         types::particle_index n_locally_owned_particles() const;
 
         /**
+         * Return a reference to the property pool that owns all particle
+         * properties, and organizes them physically.
+         */
+        PropertyPool &
+        get_property_pool() const;
+
+        /**
          * Return the number of particles in the given cell.
          */
-        unsigned int n_particles_in_cell(const typename Triangulation<dim>::active_cell_iterator &cell) const;
+        unsigned int
+        n_particles_in_cell(const typename Triangulation<dim,spacedim>::active_cell_iterator &cell) const;
+
+        /**
+         * Get a map between subdomain id and the neighbor index. In other words
+         * the returned map answers the question: Given a subdomain id, which
+         * neighbor of the current processor's domain (in terms of a contiguous
+         * number from 0 to n_neighbors) owns this subdomain?
```

Comment:
so is the value of the map then a number between zero and `n_mpi_processes(...)`? (it's not an MPI rank, I think.) maybe say this -- I had to read this sentence a few times before I understood what the function does.

By the way do you need this function to be public?